pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDaoImpl.java

```diff
@@ -492,18 +492,20 @@ public Optional<ScanError> insertKitTracking(KitRequestShipping kitRequestShippi
                     dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(),
                             "Kit Label \"" + kitRequestShipping.getKitLabel() + "\" does not exist.\n"
                                     + UserErrorMessages.IF_QUESTIONS_CONTACT_DEVELOPER);
+                    logger.error("The number of affected rows for kit label " + kitRequestShipping.getKitLabel() + "is not 1.");
                 } else {
                     logger.info("Added tracking for kit w/ kit_label " + kitRequestShipping.getKitLabel());
                 }
             } catch (Exception ex) {
                 dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(),
                         "Kit Label \"" + kitRequestShipping.getKitLabel() + "\" does not exist.\n"
                                 + UserErrorMessages.IF_QUESTIONS_CONTACT_DEVELOPER);
+                logger.error("Unable to save kit tracking information for kit label " + kitRequestShipping.getKitLabel(), ex);
             }
             return dbVals;
         });
         if (Objects.nonNull(results.resultValue)) {
-            result = (Optional<ScanError>) results.resultValue;
+            result = Optional.ofNullable((ScanError)results.resultValue);
         }
```

Comment:
My advice is to get rid of `SimpleResult dbVals` and create instead an object of type ScanError . SimpleResults is not doing anything. And you will be able to get rid of the casting.

Reply:
Thanks for the tip! I tried it. The thing is, I can't create the `ScanError` Object  before the `inTransaction` lambda and set it in the lambda and also I can't create it in the lambda and return it, since it should return `SimpleResult`.

Reply:
I see! That SimpleResult thing ought to use generics. At any rate ✅

Reply:
Thanks!